src/components/main-layout/ContentPanelTimer.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useContext, useEffect, useRef, useState } from "react";
+
+import styles from "./layout.module.css";
+import { SimulariumContext } from "../../simulation/context";
+import { PageContent, Module } from "../../types";
+
+import classNames from "classnames";
+import { isEqual } from "lodash";
+import ContentPanel from "./ContentPanel";
+
+export interface ContentPanelProps {
+    pageContent: PageContent;
+    currentModule: Module;
+}
+
+// these three states are used to avoid flashing the content before the
+// opacity transition starts
+enum RenderState {
+    NewContent = 1,
+    StartFade = 2,
+    FullyVisible = 3,
```

Comment:
I think these names aren't very clear to me... Is `NewContent` indicating that the content is hidden? Does `StartFade` mean it's fading in or fading out the content?

Reply:
how about 
option 1: `Initial` `InTransition` `Finished`. 
option 2: `NoRender` `RenderButFadingIn` `FullyVisible` 
To answer your question this is only for the fade in. 
Or some combination of those?

Reply:
I think either option would be fine... leaning towards 2 but maybe change it to `RenderFadeIn` for the second value.
I'd probably have a better idea of what the code is doing with a video or demo, but I still haven't been able to figure out my local `bun` setup yet so I wasn't able to test locally.